src/app/chat-page/chat-page.component.ts

```diff
@@ -186,6 +186,8 @@ export class ChatPageComponent implements OnInit, AfterViewInit {
             options.push(option);
           }
           const lastReport = this.storage.reports.length - 1;
+          // combine an optional street and the city
```

Comment:
I don't think this is necessary - we can use city_town and street_name here and calculate _location a few steps later.

Reply:
removed